examples/seq2seq/seq2seq_trainer.py

```diff
@@ -0,0 +1,179 @@
+import logging
+from typing import Any, Dict, Optional, Tuple, Union
+
+import torch
+from packaging import version
+from torch import nn
+from torch.utils.data import DistributedSampler, RandomSampler
+
+from transformers import Trainer
+from transformers.file_utils import is_apex_available, is_torch_tpu_available
+from transformers.trainer import get_tpu_sampler
+
+
+_use_native_amp = False
+_use_apex = False
+
+# Check if Pytorch version >= 1.6 to switch between Native AMP and Apex
+if version.parse(torch.__version__) < version.parse("1.6"):
+    if is_apex_available():
+        from apex import amp
+    _use_apex = True
+else:
+    _use_native_amp = True
+    from torch.cuda.amp import autocast
+
+try:
+    from .utils import label_smoothed_nll_loss
+except ImportError:
+    from utils import label_smoothed_nll_loss
+
+
+logger = logging.getLogger(__name__)
+
+
+class Seq2SeqTrainer(Trainer):
+    def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
+        if isinstance(self.train_dataset, torch.utils.data.IterableDataset):
+            return None
+        elif is_torch_tpu_available():
+            return get_tpu_sampler(self.train_dataset)
+        else:
+            if self.args.sortish_sampler and self.args.n_gpu <= 1 and self.args.local_rank == -1:
```

Comment:
should we `assert` here when running on multi-gpu ?

Reply:
I would 
`warnings.warn('sortish_sampler is being ignored because n_gpu > 1')`